buildSrc/src/main/groovy/org/gradle/build/docs/ExtractSnippetsTask.groovy

```diff
@@ -15,38 +15,68 @@
  */
 package org.gradle.build.docs
 
+import org.gradle.api.Action
+import org.gradle.api.DefaultTask
 import org.gradle.api.file.FileTree
 import org.gradle.api.file.FileVisitDetails
 import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.OutputDirectory
 import org.gradle.api.tasks.PathSensitive
 import org.gradle.api.tasks.PathSensitivity
-import org.gradle.api.tasks.OutputDirectory
-import org.gradle.api.tasks.SourceTask
 import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.util.PatternFilterable
 
 import java.util.regex.Matcher
 import java.util.regex.Pattern
+
 /**
  * Produces the snippets files for a set of sample source files.
  */
 @CacheableTask
-class ExtractSnippetsTask extends SourceTask {
+class ExtractSnippetsTask extends DefaultTask {
 
     @OutputDirectory
     File destDir
 
     @OutputDirectory
     File snippetsDir
 
-    @Override
+    @Internal
+    File samples
+
+    @InputFiles
     @PathSensitive(PathSensitivity.RELATIVE)
     FileTree getSource() {
-        return super.getSource()
+        def source = project.fileTree(samples)
+        source.with {
+            exclude 'userguideOutput/**'
```

Comment:
seems odd to bake this stuff into a task implementation.

Reply:
I moved this to another PR.  The task wasn't cacheable and caused us problems when we wanted to build a new variant of googletest.